### DISCUSSION TITLE
pass widget arguments to scripts called by event handlers.

### DISCUSSION THREAD

**Discussion by mantacid** [elkowar/eww#899] [Q&A] [answered]
I am having some difficulty getting variables passed in as arguments to a widget to update.
I'm trying to do something like the following:
```
(defvar testing "old_value")
(defwidget example [arg]
    (button :onclick "eww update ${arg}='new_value'")
)
```
Now, I know that wrapping the variable in those curly brackets is needed if calling it from within a string. However, when testing the widget, the variable **does not update**.

Am I doing something wrong? or can you just not evaluate variables from within an eww widget?

**Comment by mantacid** [comment]
I've thought about passing a variable as a JSON key-value pair and getting the name of the passed variable using jq, but I soon realized that eww wouldn't allow me to change a specific value of the JSON.

**Comment by w-lfchen** [comment]
```
(defvar testing "red")
(defwidget example [arg]
  (button
    :style "background-color: ${testing};"
    :onclick "eww update ${arg}='green'"
  ))

(defwindow tests
  :geometry (geometry
    :anchor "top center"
  )
  :stacking "fg"
  :windowtype "dock"
  (example :arg "testing"))
```
I don't really know what you're talking about, as the button turns green upon being pressed on my machine.
Did I miss something?

**Reply by mantacid** [reply]
I thought i learned this lesson when I learned bash, but apparently I'm still forgetting that you cant just pass a variable in as an argument you need to pass in its name. That said, I did test your example, and while the values are now updating the styles aren't showing at all.

for now I'm passing both the variable and the name of the variable as two separate arguments, so I can call whichever one is warranted by the circumstances.

**Reply by mantacid** [reply, marked as answer]
I didn’t notice this last night , but when you define the styles you call the variable directly, rather than as `${arg}`. If I replace the call to `${testing}` with a call to `${arg}` it sets the **name** of the variable (ie: “testing”) as the value in the css. In other words, the way variables get called is dependent upon the context they are called in. If I want the flexibility of being able to call a variable in any context, I would **need** to pass both the variables name and the variable itself as arguments, such that this context-dependent behavior is mitigated. I’m marking this as the solution for now, but I’m opening a bug report about this later.
